### ISSUE TITLE
Models for person_detect used in micro vision demo of Tensorflow Lite

### ISSUE THREAD

**Issue by IlkayW** [tensorflow/tensorflow#44322]
Hello,

did you upload the source models?
It would be very nice for me to get the Tensorflow Lite model and the original unquantized model file.

Thank you in advance,
Ilkay

---

They are currently only available as c++ source files.  I will have a look at adding the source model to the download.  In the meantime, a simple c++ method which dumps the array to file would re-create the tflite model.  You can find the grayscale c++ file [here](https://storage.googleapis.com/download.tensorflow.org/data/tf_lite_micro_person_data_grayscale.zip):  and the RGB model c++ file [here](https://storage.googleapis.com/download.tensorflow.org/data/tf_lite_micro_person_data.tgz).

_Originally posted by @njeffrie in https://github.com/tensorflow/tensorflow/issues/29792#issuecomment-529726207_

**Comment by bhavikapanara** [comment]
Hello @IlkayW 
Are you able to get TFLite model?

Thanks,
Bhavika

**Comment by IlkayW** [comment]
@bhavikapanara not yet unfortunately.

**Comment by njeffrie** [comment]
Here's are tflite versions for the quantized int8 and uint8 models we use in the test. In order to get a floating point model, you can try this experimental [tf2 keras training pipeline](https://github.com/mlperf/tiny/tree/master/v0.1/reference_submissions/person_detection) (accuracy is lower than the slim based [tf1 pipeline](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/person_detection/training_a_model.md), but it is easier to understand and modify).

[vww_models.zip](https://github.com/tensorflow/tensorflow/files/5484143/vww_models.zip)

**Comment by IlkayW** [comment]
@njeffrie Why doesn't the uint8 model have any activation functions?
I am using netron to inspect the model and there are no ReLU-6 activations.

**Comment by njeffrie** [comment]
In TFLite flatbuffer models, we fuse activation functions to the previous operator, so that the activation is applied directly after the bias is added in the previous layer. This is an optimization to reduce overhead by reducing the number of operators in a network. With quantized models we go one step further and implement relu variants using quantized ranges. i.e. for an operator with a relu activation after it, we set the output range to be [0, observed_max] where observed_max is set during quantized calibration. This way, the operator itself will clip the output values to the specified range without the need for any additional logic.

**Comment by njeffrie** [comment]
I have changed how we package tflite models in our examples. Now we include the quantized person detection model under `tensorflow/lite/micro/models`. I've been meaning to update the example to include full training which would help if you wanted the float model, but for now you can find the training and various models (float, quantized)  [https://github.com/mlcommons/tiny/tree/master/v0.5/training/visual_wake_words](here).
